### DISCUSSION TITLE
asiolib not found

### DISCUSSION THREAD

**Discussion by grbba** [daniele77/cli#100] [Q&A] [answered]
Hello,

again a question. I managed to successfully compile your repo on my machine with all the examples with the Boost Asio library.

I then basically took the complete.cpp code and added it into my programm and it doesn't compile anymore by complaining the asilib in server.h is not known ... 

I have an include folder in my project into which i copied the whole cli tree of includes. Any idea where that comes form ? 

Thanks a lot
//

**Comment by daniele77** [comment, marked as answer]
Be sure to include these headers before anything else:

    // if you use boost asio:
    #include <cli/boostasioscheduler.h>
    #include <cli/boostasioremotecli.h>

    // if you use standalone asio:
    #include <cli/standaloneasioscheduler.h>
    #include <cli/standaloneasioremotecli.h>

Let me know.

**Reply by grbba** [reply]
Thanks for the advice as well as all the others you send. Its very much appreciated as i am into C++ just a couple of months now so still a great deal to learn :) 

The  issue went away 👍
